### DISCUSSION TITLE
Offending whitespace characters

### DISCUSSION THREAD

**Discussion by zanodor** [SilentVoid13/Templater#1146] [Help] [answered]
For the last few weeks, the template I have been using for a long time successfully started acting up. No idea what has changed..?
This is my full script:  
````
<%-* 
if (tp.file.title.charAt(0) === "A") {
    await tp.file.move("A" + "/" + tp.file.title)
}
if (tp.file.title.charAt(0) === "Á") {
    await tp.file.move("A" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "B") {
    await tp.file.move("B" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "C") {
    await tp.file.move("C" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "D") {
    await tp.file.move("D" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "E") {
    await tp.file.move("E" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "É") {
    await tp.file.move("E" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "F") {
    await tp.file.move("F" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "G") {
    await tp.file.move("G" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "H") {
    await tp.file.move("H" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "I") {
    await tp.file.move("I" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Í") {
    await tp.file.move("I" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "J") {
    await tp.file.move("J" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "K") {
    await tp.file.move("K" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "L") {
    await tp.file.move("L" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "M") {
    await tp.file.move("M" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "N") {
    await tp.file.move("N" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "O") {
    await tp.file.move("O" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Ó") {
    await tp.file.move("O" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Ö") {
    await tp.file.move("O" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Ő") {
    await tp.file.move("O" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "P") {
    await tp.file.move("P" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Q") {
    await tp.file.move("Q" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "R") {
    await tp.file.move("R" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "S") {
    await tp.file.move("S" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "T") {
    await tp.file.move("T" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "U") {
    await tp.file.move("U" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Ú") {
    await tp.file.move("U" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Ü") {
    await tp.file.move("U" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Ű") {
    await tp.file.move("U" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "V") {
    await tp.file.move("V" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "W") {
    await tp.file.move("W" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "X") {
    await tp.file.move("X" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Y") {
    await tp.file.move("Y" + "/" + tp.file.title)
}
else if (tp.file.title.charAt(0) === "Z") {
    await tp.file.move("Z" + "/" + tp.file.title)
}
_%>
---
type: 
topics: 
tags: [formatted🟢]
publish: true
alias: [<%tp.file.title.toLowerCase()%>]
date created: <%tp.date.now("YYYY-MM-DD")%>
date modified: <%tp.date.now("YYYY-MM-DD")%>
---

# <%tp.file.title%>


````

What I get when creating a new file is five tab characters before the first `---` (1 tab = 2 spaces here) and a single whitespace where the cursor is placed at the last line.  
Not sure what causes this behavior.  
I tried appending some Templater javascript to trim the unwanted characters but couldn't, so instead I use Apply Patterns regex patterns to deal with them.  

The same behaviour happens on Windows, Linux and iOS installations.

**Comment by Zachatoo** [comment]
I am not experiencing this issue, see video below.

Some steps you can take to try to narrow down the issue:
1. Disable all plugins except the Templater plugin, restart Obsidian, and try applying the template. If it works, then it's likely another plugin you have installed that is causing the issue. You can try enabling a few plugins at a time and running your template again until the issue happens again, then you know it was one of the last few plugins you enabled.
1. Try creating a new vault, installing just the Templater plugin, and running your template. If it works, it's likely a setting in either Templater or another core Obsidian setting that is causing the issue.

https://github.com/SilentVoid13/Templater/assets/6936914/801775e9-d45c-4901-a08f-507547a31363

**Reply by zanodor** [reply]
Thanks for looking into it.  
It works in another vault even when other plugins (basically the same ones I use in my main vault) are installed. I changed some settings, disabled some stuff here and there. No change. The spaces still come up.

The reason I shared my experience is because these spaces might ring a bell with someone from other areas. I don't have a vast experience with coding and such.  
It is indeed puzzling.

**Reply by Zachatoo** [reply]
Have you tried the first step I provided? That would help you narrow down the issue systematically to which plugin specifically is causing the issue.

**Reply by zanodor** [reply]
In the TEMP vault I was using on iOS, the same themes, snippets and almost the same amount of plugins were installed and the problem does not come up. I tried disabling some of the plugins in my main vault to try and mirror the state of my TEMP vault and I couldn't make the problem go away, which means it's something else. I used different trigger folders in Templater, again, to mirror the other vault. Nada.

**Reply by Zachatoo** [reply]
This is why I made my first suggestion, so that you can know for sure if it's Templater or not, and go from there.

I can't offer any more advice until you do that step, otherwise we'll be here for weeks trying to figure out what is different between your two vaults, it's way faster to do my first suggestion.

**Reply by zanodor** [reply]
Zach, thanks but I followed every step. 
1) Installed Templater in Sandbox vault, copied over all my templates and ran a few of them, no problem.  
2) Tried in secondary (no longer used) vault with many plugins enabled and the templates I tried WERE working without these extra spaces.  
3) Tried to look for differences between primary and secondary vault and not really finding any major differences but still played around trying to replicate the other vault, to no avail. I even disabled all plugins and snippets and tried with Templater being the solely enabled plugin. No dice. I even tweaked some of the settings of Templater. Nothing.  

I tried looking for an earlier release of Templater but found that it was no updated for a long time so there is no use downgrading it. And as I said, the TEMP vault templates work.  

Interestingly, though, I have a Zotero Integration template that runs though that plugin, and it doesn't add spaces.  
All other templates run trough a meta templater (made by Pamela Wang based on Bryan Jenks I suppose) which calls a javascript script. There is no use copying those here as they are working in the other vault.  

If it was just one install on a desktop I would day it could be Obsidian, but it's all three platforms acting up and what's with those spaces anyway? How would they be generated? 

Anyway, thanks, Zach, I guess I'd like you to stop wasting your time on this. I either use a command to clean up or migrate one vault into the other to see if it'll work.  

Cheers

**Comment by zanodor** [comment, marked as answer]
I seem to have found the cause of the issue. The plugin Linter must have included my Templater md files when effecting changes -- among them soft lines breaks (two spaces at the end of each line).  
Removing the spaces from my meta template picker file made the issues go away.
